src/ai/ai_help_structs.cc

```diff
@@ -405,13 +406,13 @@ void BuildingObserver::unset_is(const BuildingAttribute attribute) {
 AiModeBuildings BuildingObserver::aimode_limit_status() const {
 	if (total_count() > cnt_limit_by_aimode) {
 		return AiModeBuildings::kLimitExceeded;
-	} else if (total_count() == cnt_limit_by_aimode) {
+	}
+	if (total_count() == cnt_limit_by_aimode) {
 		return AiModeBuildings::kOnLimit;
-	} else {
-		return AiModeBuildings::kAnotherAllowed;
 	}
+	return AiModeBuildings::kAnotherAllowed;
 }
-bool BuildingObserver::buildable(const Widelands::Player& p) {
+bool BuildingObserver::buildable(const Widelands::Player& p) const {
```

Comment:
what is the check behind this and what effect is desired.

Reply:
Making functions `const` or `static` where applicable allows you to call them from other const/static functions. You can't call a non-const/non-static function from a const/static function or on a const object.
It also guards against side-effects-induced bugs where we might accidentally (or due to a bad design choice) modify an object that is meant not to be modified.

Reply:
ok so the fuction is const now but the resulting bool isn't - correct?

Reply:
Correct. The return value is passed by value here (that is, it's not a pointer or reference) and such values actually *cannot* be const/volatile-qualified. There's even another clang-tidy check that points out that `const bool some_function()` doesn't make sense.